packages/js/components/src/experimental-select-control/stories/index.tsx

```diff
@@ -162,30 +162,80 @@ export const Async: React.FC = () => {
 
 export const CustomRender: React.FC = () => {
 	const [ selected, setSelected ] = useState< DefaultItemType[] >( [] );
+	const [ isOpen, setIsOpen ] = useState( false );
 
 	const onRemove = ( item ) => {
 		setSelected( selected.filter( ( i ) => i !== item ) );
 	};
 
 	const onSelect = ( item ) => {
-		const isSelected = selected.find( ( i ) => i === item );
+		const isSelected = selected.find( ( i ) => i.value === item.value );
 		if ( isSelected ) {
 			onRemove( item );
 			return;
 		}
 		setSelected( [ ...selected, item ] );
 	};
 
+	const getFilteredItems = (
+		allItems: DefaultItemType[],
+		inputValue: string,
+		selectedItems: DefaultItemType[],
+		getItemLabel: getItemLabelType< DefaultItemType >
+	) => {
+		const escapedInputValue = inputValue.replace(
+			/[.*+?^${}()|[\]\\]/g,
+			'\\$&'
+		);
+		const re = new RegExp( escapedInputValue, 'gi' );
+
+		return allItems.filter( ( item ) => {
+			return re.test( getItemLabel( item ).toLowerCase() );
+		} );
+	};
+
 	return (
 		<>
 			<SelectControl
 				multiple
 				label="Custom render"
 				items={ sampleItems }
-				getFilteredItems={ ( allItems ) => allItems }
 				selected={ selected }
 				onSelect={ onSelect }
 				onRemove={ onRemove }
+				getFilteredItems={ getFilteredItems }
+				isOpen={ isOpen }
+				onStateChange={ (
+					{ type, selectedItem },
+					{ setInputValue, setIsFocused, selectItem }
+				) => {
+					switch ( type ) {
+						case selectControlStateChangeTypes.InputBlur:
+						case selectControlStateChangeTypes.InputKeyDownEscape:
+							setInputValue( '' );
+							setIsOpen( false );
+							setIsFocused( false );
+							break;
+						case selectControlStateChangeTypes.InputChange:
+						case selectControlStateChangeTypes.InputKeyDownArrowDown:
+						case selectControlStateChangeTypes.InputKeyDownArrowUp:
+							if ( ! isOpen ) {
+								setIsOpen( true );
+							}
+							break;
+						case selectControlStateChangeTypes.InputKeyDownEnter:
+						case selectControlStateChangeTypes.FunctionSelectItem:
+						case selectControlStateChangeTypes.ItemClick:
```

Comment:
I probably should have looked this up earlier, but I think the correct action on selection is to close the list box and not keep it open.  https://www.w3.org/WAI/ARIA/apg/patterns/combobox/#listbox-popup-keyboard-interaction

I'm not sure if leaving the box open on click and only close it when using the keyboard satisfies the accessibility concerns.  But I don't want to hold you up on this PR and we can address later; there are other good changes worth merging in this PR.